mozilla/fx-private-relay-add-on
# [MPP-1714] Popup Refactor - Panel Navigation and Header Bar

## Summary

This is the first in a series of PRs overhauling the add-on popup. 

Links: 
- Jira Epic: https://mozilla-hub.atlassian.net/browse/MPP-1708
- [Figma designs](https://www.figma.com/file/NP5tLeIFJeHC5HvaqpLQdp/MPPUX-129-Add-On?node-id=144%3A5071&t=MN61ZFZ9w2429nNj-4) 

Work completed so far: 
- Stub new work area for toolbar
- Add new navigation icons
- Add hard-coded (EN) labels for navigation
- Start popup javascript logic (including init, panel switching, and login detection)

TODO:
- [x] Make sign-in links on popup work as expected
- [x] Make "Manage"/external link work correctly 

## Tests

1. Confirm correct panel loads if user is logged in/out
- Run the add-on
- Open pop-up (before logging in) 
- Expected: Panel should display "Login" content (_"Get started by signing in to Relay with your Firefox account."_) 
- Log in using First Run page
  - Note: The login content in the popup does not work currently
- Open pop-up again
- Expected: Panel should display "MASKS_PANEL" text. 

2. Confirm if user logs out of Relay, the add-on requires user to log back in
- Run the add-on
- Log in with FxA
- Log out from Relay website
- Open panel
- Expected: Login content (unstyled) should be visible

3. Confirm external link (Manage) works
- Run the add-on
- Click the "external link/Manage" (arrow pointing out of box) icon in the navigation bar of the popup
- Expected: Relay website should open (to FxA login) and pop-up should close

4. Confirm internal navigation links change panels
- Run the add-on
- Log in
- Click on the different navigation items (except the external link/Manage)
- Expected: Each click should update the content of the popup panel
